From 06834d77798447e07fd7b36dbac6f8df14de914e Mon Sep 17 00:00:00 2001
From: William Horvath <william@horvath.blog>
Date: Sun, 29 Jun 2025 09:31:03 -0700
Subject: [PATCH 29/46] ntdll: Null-terminate unix_name in
 nt_to_unix_file_name.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Otherwise we can print garbage data like L"Reference" not found in .'���'��w
---
 dlls/ntdll/unix/file.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/ntdll/unix/file.c b/dlls/ntdll/unix/file.c
index f1da6fafe98..84b5a4185b4 100644
--- a/dlls/ntdll/unix/file.c
+++ b/dlls/ntdll/unix/file.c
@@ -3853,6 +3853,7 @@ static NTSTATUS nt_to_unix_file_name_internal( OBJECT_ATTRIBUTES *attr, UNICODE_
     unix_len = name_len * 3 + MAX_DIR_ENTRY_LEN + 3;
     if (!(unix_name = malloc( unix_len ))) return STATUS_NO_MEMORY;
     unix_name[0] = '.';
+    unix_name[1] = '\0';
 
     if (!(status = server_get_unix_fd( attr->RootDirectory, 0, &root_fd, &needs_close, &type, NULL )))
     {
-- 
2.53.0

